probe-rs/src/bin/probe-rs/main.rs

```diff
@@ -173,6 +179,37 @@ fn default_logfile_location() -> Result<PathBuf> {
     Ok(log_path)
 }
 
+/// Prune all old log files in the `directory`.
+fn prune_logs(directory: &Path) -> Result<(), anyhow::Error> {
+    // Get the path and elapsed creation time of all files in the log directory that have the '.log'
+    // suffix.
+    let mut log_files = fs::read_dir(directory)?
+        .filter_map(|entry| {
+            if let Ok(entry) = entry {
+                let path = entry.path();
+                if path.extension() == Some(OsStr::new("log")) {
+                    let metadata = fs::metadata(&path).ok()?;
+                    let last_modified = metadata.created().ok()?;
+                    Some((path, last_modified))
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect_vec();
```

Comment:
I find this a bit of an unnecessary method, since just `log_files: Vec<_>` and a simple collect would work as well. But we're already using itertools. so no harm done.

Reply:
I love this method :D But pulling an extra dependency just for it would be a waste, I agree :)